data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorConfig.java

```diff
@@ -5,50 +5,90 @@
 package org.opensearch.dataprepper.plugins.lambda.processor;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 import jakarta.validation.Valid;
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 import jakarta.validation.constraints.Size;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
 import org.opensearch.dataprepper.plugins.lambda.common.config.AwsAuthenticationOptions;
 import org.opensearch.dataprepper.plugins.lambda.common.config.BatchOptions;
-import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.BATCH_EVENT;
+import org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_CONNECTION_RETRIES;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_SDK_TIMEOUT;
 import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.REQUEST_RESPONSE;
 
-public class LambdaProcessorConfig {
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
 
-    private static final int DEFAULT_CONNECTION_RETRIES = 3;
+public class LambdaProcessorConfig {
+    //Ensures 1:1 mapping of events input to lambda and response from lambda
+    public static final String STRICT = "strict";
+    //When #input events to lambda is not equal to the response from lambda
+    public static final String AGGREGATE = "aggregate";
 
     @JsonProperty("aws")
     @NotNull
     @Valid
     private AwsAuthenticationOptions awsAuthenticationOptions;
 
+    @JsonPropertyDescription("Lambda Function Name")
     @JsonProperty("function_name")
     @NotEmpty
     @Size(min = 3, max = 500, message = "function name length should be at least 3 characters")
     private String functionName;
 
+    @JsonPropertyDescription("Total retries we want before failing")
     @JsonProperty("max_retries")
     private int maxConnectionRetries = DEFAULT_CONNECTION_RETRIES;
 
+    @JsonPropertyDescription("invocation type defines the way we want to call lambda function")
     @JsonProperty("invocation_type")
     private String invocationType = REQUEST_RESPONSE;
 
-    @JsonProperty("payload_model")
-    private String payloadModel = BATCH_EVENT;
+    @JsonPropertyDescription("sdk timeout defines the time sdk maintains the connection to the client before timing out")
+    @JsonProperty("sdk_timeout")
+    private Duration sdkTimeout = DEFAULT_SDK_TIMEOUT;
+
+    @JsonPropertyDescription("mode defines the way dataprepper treats the response from lambda")
+    @JsonProperty("response_processing_mode")
```

Comment:
I think we can drop "mode" from here as an enum is somewhat of a "mode" and we don't use that term much. So it could be `response_processing`.

Is there a better way to express this beyond "response processing?" That is, could this be expressed by the terms _cardinality_ or _count_? The heart of this configuration is to decide how the processor handles the response items.

Reply:
how about response_cardinality ?

Reply:
That sounds good to me.